### PR TITLE
Update test for latest rlang

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/warning.R [hunk 1]
```diff
@@ -70,6 +70,8 @@ print_trace <- function(cnd, ..., simplify = c("branch", "collapse", "none")) {
   trace <- cnd$trace
 
   if (!is_null(trace)) {
+    simplify <- arg_match(simplify)
```
Comment (hadley) [inline R/warning.R:73]: This ensures the code keeps working even if the possible args to `simplify` change again in the future.

Comment (lionel-) [reply R/warning.R:73]: Sorry I don't understand. Doesn't this prevent extension of `simplify` in rlang? I.e. unknown options will cause an error here.

Comment (hadley) [reply R/warning.R:73]: Another approach would be:

```R
    if (missing(simplify)) {
      cat_line(red(format(trace, ...)))
    } else {
      cat_line(red(format(trace, ..., simplify = simplify)))
    }
```

Comment (hadley) [reply R/warning.R:73]: This feels maybe like an issue that `arg_match()` should be designed to solve, but I don't see how unless we iterate up the promise stack to find if it's ever missing.

Comment (hadley) [reply R/warning.R:73]: Anyway, I'm going to merge this PR to unblock the others, but I'm happy to revisit once you get a chance to think about it more.

**Pull request merged**